src/topology/category/Profinite/cofiltered_limit.lean

```diff
@@ -110,4 +114,129 @@ begin
+  let ι : α → α → fin 2 := λ x y, if x = y then 0 else 1,
+  let ff := (f.map ι).flip,
+  have hff := λ (a : α), exists_locally_constant_fin_two _ hC (ff a),
+  choose j g h using hff,
+  let G : finset J := finset.univ.image j,
+  obtain ⟨j0,hj0⟩ := is_cofiltered.inf_objs_exists G,
+  have hj : ∀ a, j a ∈ G,
+  { intros a,
+    simp [G] },
+  let fs : Π (a : α), j0 ⟶ j a := λ a, (hj0 (hj a)).some,
+  let gg : α → locally_constant (F.obj j0) (fin 2) := λ a, (g a).comap (F.map (fs _)),
+  let ggg := locally_constant.unflip gg,
+  refine ⟨j0, ggg, _⟩,
+  have : f.map ι = locally_constant.unflip (f.map ι).flip, by simp,
+  rw this, clear this,
+  have : locally_constant.comap (C.π.app j0) ggg =
+    locally_constant.unflip (locally_constant.comap (C.π.app j0) ggg).flip, by simp,
+  rw this, clear this,
+  congr' 1,
+  ext1 a,
+  change ff a = _,
+  rw h,
+  dsimp [ggg, gg],
+  ext1,
+  repeat {
+    rw locally_constant.coe_comap,
+    dsimp [locally_constant.flip, locally_constant.unflip] },
+  { congr' 1,
+    change _ = ((C.π.app j0) ≫ (F.map (fs a))) x,
+    rw C.w },
+  all_goals { continuity },
+end
+
+theorem exists_locally_constant_fintype_nonempty {α : Type*} [fintype α] [nonempty α]
+  (f : locally_constant C.X α) :
+  ∃ (j : J) (g : locally_constant (F.obj j) α), f = g.comap (C.π.app _) :=
+begin
+  inhabit α,
+  obtain ⟨j,gg,h⟩ := exists_locally_constant_fintype_aux _ hC f,
+  let ι : α → α → fin 2 := λ a b, if a = b then 0 else 1,
+  let σ : (α → fin 2) → α := λ f, if h : ∃ (a : α), ι a = f then h.some else arbitrary _,
+  refine ⟨j, gg.map σ, _⟩,
+  ext,
+  rw locally_constant.coe_comap _ _ (C.π.app j).continuous,
+  dsimp [σ],
+  have h1 : ι (f x) = gg (C.π.app j x),
+  { change f.map (λ a b, if a = b then (0 : fin 2) else 1) x = _,
+    rw [h, locally_constant.coe_comap _ _ (C.π.app j).continuous] },
+  have h2 : ∃ a : α, ι a = gg (C.π.app j x) := ⟨f x, h1⟩,
+  rw dif_pos h2,
+  apply_fun ι,
+  { rw h2.some_spec,
+    exact h1 },
+  { intros a b hh,
+    apply_fun (λ e, e a) at hh,
+    dsimp [ι] at hh,
+    rw if_pos rfl at hh,
+    split_ifs at hh with hh1 hh1,
+    { exact hh1.symm },
+    { exact false.elim (bot_ne_top hh) } }
+end
+
+/-- Any locally constant function from a cofiltered limit of profinite sets factors through
+one of the components. -/
+theorem exists_locally_constant {α : Type*} (f : locally_constant C.X α) :
+  ∃ (j : J) (g : locally_constant (F.obj j) α), f = g.comap (C.π.app _) :=
+begin
+  let S := f.discrete_quotient,
+  let ff : S → α := f.lift,
+  by_cases hα : nonempty S,
+  { resetI,
+    let f' : locally_constant C.X S := ⟨S.proj, S.proj_is_locally_constant⟩,
+    obtain ⟨j,g',h⟩ := exists_locally_constant_fintype_nonempty _ hC f',
+    use j,
+    refine ⟨⟨ff ∘ g', g'.is_locally_constant.comp _⟩,_⟩,
+    ext1 t,
+    apply_fun (λ e, e t) at h,
+    rw locally_constant.coe_comap _ _ (C.π.app j).continuous at h ⊢,
+    dsimp at h ⊢,
+    rw ← h,
+    refl },
+  { suffices : ∃ j : J, ¬ nonempty (F.obj j),
```

Comment:
There's now a definition of `is_empty` in mathlib, it would be nice if we can propagate it to new additions

Reply:
I'm not sure this change is worth it in this case. First, the "not nonempty" thing appears only in the middle of a tactic block. Second, because I'm using `rw ← not_forall` it ends up being somewhat cumbersome to use `is_nonempty`, as this wold require rewriting inside of an existential statement, which seems to only work with `conv`. Finally, using `is_empty` would introduce another `resetI`.